integration-commons/src/main/java/org/symphonyoss/integration/parser/ParserUtils.java

```diff
@@ -123,15 +140,18 @@ public static String markupLinks(String message) {
         int endIndex = m.end();
         int stringLength = message.length();
 
+        String linkMarkup =
+            buildMarkupLinks ? "<a href=\"" + finalUrl + "\">" + finalUrl + "</a>" : finalUrl;
+        parameters.add(linkMarkup);
+
         //Rebuilds string
         message =
-            message.substring(0, startIndex) + "<a href=\"" + finalUrl + "\">" + finalUrl + "</a>"
-                + message.substring(endIndex, stringLength);
+            message.substring(0, startIndex) + "%s" + message.substring(endIndex, stringLength);
```

Comment:
There is a risk of collision here, once your message could already have a '%s' inside it. A better solution could be the usage of a LinkedHashMap, usgin a UID as key and the link as value. Then you replace them all after.